pymc/tests/distributions/test_multivariate.py

```diff
@@ -1475,14 +1472,39 @@ def test_zsn_fail_axis(self, dims, zerosum_axes):
                 with pm.Model(coords=COORDS) as m:
                     _ = pm.ZeroSumNormal("v", dims=dims, zerosum_axes=zerosum_axes)
 
-    def test_zsn_change_dist_size(self):
-        base_dist = pm.ZeroSumNormal.dist(shape=(4, 9))
+    @pytest.mark.parametrize(
+        "zerosum_axes",
+        [(-1), (-2), (1), ((0, 1)), ((-2, -1))],
+    )
+    def test_zsn_change_dist_size(self, zerosum_axes):
+        base_dist = pm.ZeroSumNormal.dist(shape=(4, 9), zerosum_axes=zerosum_axes)
+        random_samples = pm.draw(base_dist, draws=100)
+
+        if not isinstance(zerosum_axes, (list, tuple)):
+            zerosum_axes = [zerosum_axes]
+        self.assert_zerosum_axes(random_samples, zerosum_axes)
 
         new_dist = change_dist_size(base_dist, new_size=(5, 3), expand=False)
         assert new_dist.eval().shape == (5, 3)
+        random_samples = pm.draw(new_dist, draws=100)
+        self.assert_zerosum_axes(random_samples, zerosum_axes)
 
         new_dist = change_dist_size(base_dist, new_size=(5, 3), expand=True)
         assert new_dist.eval().shape == (5, 3, 4, 9)
+        random_samples = pm.draw(new_dist, draws=100)
+        self.assert_zerosum_axes(random_samples, zerosum_axes)
+
+    def assert_zerosum_axes(self, random_samples, zerosum_axes):
```

Comment:
You shouldn’t treat the negative and positive axis differently here. You should just loop across the zero sum axes and try `random_samples.mean(axis=ax)`. If that isn’t close to zero, then there’s a problem. The fact that you had to sum 1 to the positive axes to me means that something fishy is happening

Reply:
No I don't think that's a problem here: I had to add 1 because `pm.draw` adds an axis to the left for the draws dimension. So, when the axes are specified with positive integers, you have to account for it. 
Same thing when I test the NUTS samples (I had to add 2 because of `chain` and `draw` added to the left)

Reply:
Outdated: `zerosum_axes` are now to the rightmost positions, so what counts is how many there are, not _where_ they are